src/EnergyPlus/IceThermalStorage.cc

```diff
@@ -609,11 +606,15 @@ namespace IceThermalStorage {
                     AvgFracCharged = DetIceStor(IceNum).IceFracRemaining + (ChargeFrac / 2.0);
                 }
 
-                if (DetIceStor(IceNum).ChargeCurveTypeNum == CurveQuadraticLinear) {
+                // TODO: This is bad. There is no documentation for the change in meaning behind these two curve types
```

Comment:
@mjwitte Note this comment here. This happens in four places for Detailed Ice Storage. Apparently the independent variables for these curves change depending on whether it's cubic-linear or quadratic-linear. They have one independent variable in common, but it isn't in the same order. An explanation of the independent variables isn't clear in the documentation (the cubic-linear curve variables aren't mentioned at all!). There is a single example file demonstrating the cubic-linear curves, but nothing outside of the source code that explains how they are interpreted.

Reply:
@mjwitte Wanted to make sure you saw this comment, too.

Reply:
I would undo all of the changes made here for this object.  Keep the exisiting curve type fields which set 'DetIceStor(IceNum).DischargeCurveTypeNum` and `DetIceStor(IceNum).ChargeCurveTypeNum` and leave all that logic as-is in the simulation section.  In the IDD allow any 2-dimensional curve, and check for that in `GetIceStorgageInput.` 

Then post a new issue for @RKStrand to change the keys for these two curve type fields and beef up the I/O Ref documentation to clarify the difference in independent variables. The keys for these fields could something like "ChargeFractionLMTD"/"LMTDMassFlow" and "DischargeFractionLMTD"/"LMTDMassFlow". There's precedent for this elsewhere like Chiller:Electric:ReformulatedEIR (Electric Input to Cooling Output Ratio Function of Part Load Ratio Curve Type = LeavingCondenserWaterTemperature or Lift) I wouldn't tackle that here, that should be a separate issue to give it some visibility.

Reply:
I added a new issue #6879. I'll revert this here.